### DISCUSSION TITLE
Newlines not being converted to spaces within a component

### DISCUSSION THREAD

**Discussion by mgalgs** [babel/babel#13817] [Q&A] [answered]
I'm having issues with whitespace when I break up long paragraphs in my source code.

I know that adjacent components have all whitespace removed between them, e.g.:

```jsx
<div>
  <span>I like pie</span>
  <span>and pizza</span>
<div>
```

Renders as:

```
I like pieand pizza
```

I get that, no problem. We can fix that by adding a `{' '}` between the `<span>`s.

But my issue is with whitespace handling *within* a component, e.g.:

```jsx
<p>
  I like pie
  and pizza
</p>
```

In regular html this renders with the line breaks replaced with a single space, i.e.:

```
I like pie and pizza
```

But when I do this in jsx, it renders with all whitespace between lines removed, i.e.:

```
I like pieand pizza
```

My current workaround is to put `{' '}` between each line, but that's ugly and cumbersome (especially for long paragraphs):

```jsx
<p>
  I like pie
  {' '}
  and pizza
</p>
```

Assuming [this](https://github.com/facebook/react/pull/480#issuecomment-31296039) description of whitespace rules is correct, which states:

> All lines having leading or trailing whitespace are trimmed, all newlines are removed, **adjacent text separated by newlines become separated by a single space**. Any whitespace tabs are replaced with spaces. Strings inside expressions are unaffected.

I would have expected to be able to break up my long paragraph *without* having to insert `{' '}` between each line. But is seems like the "adjacent text separated by newlines become separated by a single space" part isn't happening for me.

Is this the expected behavior or have I misconfigured something? Thanks!

**Comment by nicolo-ribaudo** [comment, marked as answer]
Can you share your config, or even better an example which shows the bug? In the repl it's correctly keeping a space between `pie` and `and`: https://babeljs.io/repl#?browsers=IE%2011&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=DwBwfAUABFCSUBsCWBrAplES3SgQwDsATTJALzLwmAHpwg&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=react&prettier=false&targets=&version=7.15.7&externalPlugins=&assumptions=%7B%7D

**Reply by mgalgs** [reply]
Well, this is embarassing... My line was actually ending in a `{variable}`, i.e.:

```jsx
<p>
  I like {desert}
  and pizza
</p>
```

It totally works when the line doesn't end with a code block. No idea what I was thinking.
:facepalm: :facepalm: :facepalm: 

Sorry for the noise and thank you for the quick response!!!
